pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/with-webkit.nix

```diff
@@ -0,0 +1,34 @@
+{
+  stdenv,
+  lib,
+  telegram-desktop,
+  glib-networking,
+  webkitgtk_4_1,
+  makeBinaryWrapper,
+}:
+
+if stdenv.hostPlatform.isLinux then
+  stdenv.mkDerivation {
+    pname = "${telegram-desktop.pname}-with-webkit";
+    version = telegram-desktop.version;
+    nativeBuildInputs = [ makeBinaryWrapper ];
+    dontUnpack = true;
+    installPhase = ''
+      mkdir -p $out
+      cp -r ${telegram-desktop}/share $out
+      substituteInPlace $out/share/dbus-1/services/* --replace-fail ${telegram-desktop} $out
+    '';
+    postFixup = ''
+      mkdir -p $out/bin
+      makeBinaryWrapper {${telegram-desktop},$out}/bin/${telegram-desktop.meta.mainProgram} \
+        --inherit-argv0 \
+        --prefix GIO_EXTRA_MODULES : ${glib-networking}/lib/gio/modules \
```

Comment:
Is it possible to use `wrapGAppsHook` here?

Reply:
Yeah, `''${gappsWrapperArgs[@]}`should theoretically work just fine, it's just there seem to be no other arguments relevant for webkitgtk. Do you want gappsWrapperArgs to be used instead?

Reply:
That would be great if we don't have to manual construct these arguments.